### DISCUSSION TITLE
Async, upsert get_or_create, helpers

### DISCUSSION THREAD

**Discussion by tuxpowered** [fastapi/sqlmodel#1930] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
# generic question, no code sample.
```

### Description

I understand this is still a very very very new project, and that the current documentation is not as well vetted as FastAPI is. So if the following exists, maybe someone can point me to documentation or example and maybe I can help write it up.

1. It is not clear if `sqlmodel` is async aware or how to use it if it is.
2. How is sqlmodel different than say ORMAR which __is__ async aware and uses a single shared pydantic model?
3. It appears that common helper functions of other ORMs such as `upsert` and `get_or_create` are absent from `sqlmodel`,  is there a plan to add native support for these options?

### Operating System

Linux, macOS

### SQLModel Version

0.0.4

### Python Version

3.9.5

**Comment by perfecto25** [comment]
I was able to run in async using this tutorial

https://testdriven.io/blog/fastapi-sqlmodel/

**Comment by perfecto25** [comment]
this repo shows a good way to create an async session

https://github.com/Cedric-Magnan/fastapi-async-sqlmodel/blob/main/app.py

**Comment by iloveitaly** [comment]
👋 I was looking for the exact same helpers, so I built them into a [wrapper library](https://github.com/iloveitaly/activemodel/)! It adds `find_or_create_by()`, `upsert()`, and other rails-like features. It also includes an ORM-like DX, lifecycle hooks, easy session management, and JSONB mutation tracking. Check it out! Would love to hear any feedback and have more contributions to it.
